### PR TITLE
feat: deprecate mergeOptions() and use merge() instead

### PR DESCRIPTION
**Pull request by gjanblaszczyk**
## Description
The video.js mergeOptions() function is deprecated. The plugin should use videojs.obj.merge() function.

## Specific Changes proposed
- remove videojs.mergeOptions() and use videojs.obj.merge() instead
- fix js-doc warnings
- remove old videojs.plugin

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [x] If necessary, more likely in a feature request than a bug fix
  - [ ] Unit Tests updated or fixed
  - [ ] Docs/guides updated
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

**Comment by gjanblaszczyk** [comment]
Maybe I should also update video.js version in package.json from "6 | 7" to "8". @gkatsev any thoughts?

**Comment by gkatsev** [comment]
@gjanblaszczyk I was just thinking about this that this would be a breaking change but maybe not an issue. Yeah, if we want to keep it as a breaking change, we should update the package.json. Alternative is to use `videojs.obj.merge` if available and otherwise use `videojs.mergeOptions()`.

**Comment by gjanblaszczyk** [comment]
@gkatsev I removed v6 from package.json. Is that fine? or maybe It is still needed...

**Comment by gkatsev** [comment]
Yeah, that would work, though, technically still a breaking change.

File: package.json [hunk 1]
```diff
@@ -58,10 +58,10 @@
   ],
   "dependencies": {
     "global": "^4.3.2",
-    "video.js": "^6 || ^7"
+    "video.js": "^7 || ^8"
```
Comment (misteroneill) [inline package.json]: I believe this should still be compatible with `^6` now that we fall back to `mergeOptions`, so we might as well keep it in there.

But I think #142 is probably a big enough change to warrant a major version of this plugin, so maybe the best move is just to bite the bullet and make it compatible with 8+ only and increment the major version?

**Review by misteroneill — CHANGES_REQUESTED** [review]
Yeah, let's just do a breaking change here. @gjanblaszczyk would you mind making this compatible with `^8` only and removing the `mergeOptions` fallback?

Let's keep it simple!

**Comment by gjanblaszczyk** [comment]
@misteroneill Sure, I don't have a problem with that. I have just updated the PR. Does it look fine?

File: package.json (hunk 1 above)
Comment (gjanblaszczyk) [reply package.json]: Yeah, It sounds good!

File: src/quality-level-list.js [hunk 2]
```diff
@@ -103,7 +106,7 @@ class QualityLevelList extends videojs.EventTarget {
   /**
    * Removes a quality level from the list.
    *
-   * @param {QualityLevel} remove QualityLevel to remove to the list.
+   * @param {QualityLevel} qualityLevel QualityLevel to remove to the list.
```
Comment (mister-ben) [inline src/quality-level-list.js]:
```suggestion
   * @param {QualityLevel} qualityLevel  The QualityLevel to remove from the list.
```

Comment (gjanblaszczyk) [reply src/quality-level-list.js]: Thanks @mister-ben. I have just updated the PR.

**Review by misteroneill — APPROVED** [review]

**Pull request merged**